src/Database/PostgreSQL/Simple/Internal.hs

```diff
@@ -82,6 +85,23 @@ data Connection = Connection {
 instance Eq Connection where
    x == y = connectionHandle x == connectionHandle y
 
+-- | Superclass for postgresql exceptions
+data SomePostgreSqlException = forall e. Exception e => SomePostgreSqlException e
```

Comment:
I don't understand. This is exactly the same as `SomeException`. This type doesn't seem to provide any additional guarantees?

Reply:
Ah, I see. This is the same "trick" as `SomeAsyncException`.